src/System.Threading/tests/ThreadLocalTests.cs

```diff
@@ -0,0 +1,418 @@
+            bool successful = true;
+            string values = "";
+            for (int i = 1; i <= threads.Length; ++i)
+            {
+                string seenValue = seenValuesFromAllThreads[i - 1];
+                values += seenValue + ",";
+                if (seenValue != i.ToString())
+                {
+                    successful = false;
+                }
+            }
+
+            if (!successful)
+            {
+                Debug.WriteLine("RunThreadLocalTest4_Value: > test failed - ThreadLocal test failed. Seen values are: " + values.Substring(0, values.Length - 1));
+                Assert.True(false, string.Format("RunThreadLocalTest4_Value: > test failed - ThreadLocal test failed. Seen values are: " + values.Substring(0, values.Length - 1)));
+            }
+        }
+
+        [Fact]
+        public static void RunThreadLocalTest4_Value_NegativeCases()
+        {
+            ThreadLocal<string> tlocal = null;
+            try
+            {
+                int x = 0;
+                tlocal = new ThreadLocal<string>(delegate
+                {
+                    if (x++ < 5)
+                        return tlocal.Value;
+                    else
+                        return "Test";
+                });
+                string str = tlocal.Value;
+                Assert.True(false, string.Format("RunThreadLocalTest4_Value: > test failed - expected exception InvalidOperationException"));
+            }
+            catch (InvalidOperationException)
+            {
+            }
+        }
+
+        [Fact]
+        public static void RunThreadLocalTest5_Dispose()
+        {
+            // test recycling the combination index;
+            ThreadLocal<string> tl = new ThreadLocal<string>(() => null);
+            if (tl.IsValueCreated)
+            {
+                Assert.True(false, string.Format("RunThreadLocalTest5_Dispose: Failed: IsValueCreated expected to return false."));
+            }
+            if (tl.Value != null)
+            {
+                Assert.True(false, string.Format("RunThreadLocalTest5_Dispose: Failed: reusing the same index kept the old value and didn't use the new value."));
+            }
+
+            // Test that a value is not kept alive by a departed thread
+            var threadLocal = new ThreadLocal<SetMreOnFinalize>();
+            var mres = new ManualResetEventSlim(false);
+
+            // (Careful when editing this test: saving the created thread into a local variable would likely break the 
+            // test in Debug build.)
+            // We are creating the task using TaskCreationOptions.LongRunning because...
+            // there is no guarantee that the Task will be created on another thread.
+            // There is also no guarantee that using this TaskCreationOption will force
+            // it to be run on another thread.
+            new Task(() => { threadLocal.Value = new SetMreOnFinalize(mres); }, TaskCreationOptions.LongRunning).Start();
```

Comment:
Same as the other new Task(..., LongRunning).Start()... either it should use Task.Factory.StartNew(..., CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default) or TaskScheduler.Default should be passed to Start(TaskScheduler.Default).